script/make-firmware.bash

```diff
@@ -29,98 +29,303 @@
+            ln -s ../openthread .
+
+            # Build
+            build_dir=${OT_CMAKE_BUILD_DIR:-"${repo_dir}"/build-"${thread_version}"/"${platform}"}
+            options=("${build_1_3_options_common[@]}")
+            case "${platform}" in
+                nrf*)
+                    options+=("${build_1_3_options_nrf[@]}")
+                    ;;
+                efr32mg12)
+                    options+=("-DBOARD=brd4166a" ${build_1_3_options_efr32[@]})
+                    ;;
+            esac
+            OT_CMAKE_BUILD_DIR=${build_dir} ./script/build ${platform} ${build_type:-""} "${options[@]}" "$@"
+
+            # Package and distribute
+            local dist_apps=(
+                ot-cli-ftd
+                ot-rcp
+            )
+            for app in ${dist_apps[@]}; do
+                package "${app}"
+            done
+
+            # Clean up
+            rm -rf openthread
+            git clean -xfd
+            git submodule update --force
+            ;;
+
+        # Build OpenThread 1.2
+        "1.2")
+            cd ${platform_repo}
+            git clean -xfd
+
+            # Use OpenThread from top-level of repo
+            rm -rf openthread
+            ln -s ../openthread .
+
+            # Build
+            build_dir=${OT_CMAKE_BUILD_DIR:-"${repo_dir}"/build-"${thread_version}"/"${platform}"}
+            options=("${build_1_2_options_common[@]}")
+            case "${platform}" in
+                nrf*)
+                    options+=("${build_1_2_options_nrf[@]}")
+                    ;;
+            esac
+            OT_CMAKE_BUILD_DIR=${build_dir} ./script/build ${platform} ${build_type:-""} "${options[@]}" "$@"
+
+            # Package and distribute
+            local dist_apps=(
+                ot-cli-ftd
+                ot-rcp
+            )
+            for app in ${dist_apps[@]}; do
+                package "${app}"
+            done
+
+            # Clean up
+            rm -rf openthread
+            git clean -xfd
+            git submodule update --force
+            ;;
+
+        # Build OpenThread 1.1
+        "1.1")
+            cd openthread-1.1
+
+            # Prep
+            git clean -xfd
+            ./bootstrap
+
+            # Build
+            options=("${build_1_1_env_common[@]}")
+            case "${platform}" in
+                nrf*)
+                    options+=("${build_1_1_env_nrf[@]}")
+                    ;;
+            esac
+            make -f examples/Makefile-${platform} "${options[@]}" "$@"
+
+            # Package and distribute
+            local dist_apps=(
+                ot-cli-ftd
+                ot-rcp
+            )
+            for app in ${dist_apps[@]}; do
+                package ${app} ${thread_version} output/${platform}/bin/${app}
+            done
+
+            # Clean up
+            git clean -xfd
+            ;;
+    esac
+
+    cd ${repo_dir}
+}
+
+die() { echo "$*" 1>&2 ; exit 1; }
+
+main()
+{
+    if [[ $# == 0 ]]; then
```

Comment:
For the same reasons we are now allowing make-reference-release.bash to be run without args I think we should do the same thing for this script.

Reply:
I disagree. This script is mainly used as a part of `./script/make-reference-release`. Having this script take in the platform as an argument allows for easier debugging/testing for when you only want to build one platform.

@suveshpratapa thoughts?

Reply:
I'm ok with allowing for a platform specification. I'm just saying we shouldn't require one.

Reply:
@lmnotran Are you implying this script is not called anywhere else, and in its invocation always takes an argument? If so that's okay.

However, we shouldn't break any existing functionality, and if there's a use case for it to be called without an argument (therefore building for all platforms), we should keep it in.

Reply:
@suveshpratapa 
> @lmnotran Are you implying this script is not called anywhere else, and in its invocation always takes an argument? If so that's okay.

Yup! That's exactly what I'm saying. It's only called https://github.com/SiliconLabs/ot-reference-release/pull/1/files#diff-df3647cf1ac7f622c7a4f57f9762626a7e84d775dc4e7209aa7a013e85d0746dR47-R49

@romacdon 
> I'm ok with allowing for a platform specification. I'm just saying we shouldn't require one.

I can make it so that the `make-firmware.bash` builds all firmware when no args are given and only builds a specific platform if one is given.

Reply:
> > I'm ok with allowing for a platform specification. I'm just saying we shouldn't require one.
> 
> I can make it so that the `make-firmware.bash` builds all firmware when no args are given and only builds a specific platform if one is given.

Great -= Thanks

Reply:
@romacdon Actually, I don't think doing that would the cleanest solution. We would need to loop through pretty much the entire body of `make-firmware.bash: main()` for each platform.

Can we just leave it as is and ask the OpenThread community for their input?

Reply:
> @romacdon Actually, I don't think doing that would the cleanest solution. We would need to loop through pretty much the entire body of `make-firmware.bash: main()` for each platform.
> 
> Can we just leave it as is and ask the OpenThread community for their input? From the perspective of the main use-case, this doesn't break anything. See https://github.com/SiliconLabs/ot-reference-release/blob/main/README.md

My concern is that you are changing the interface.  Yes, make-reference-release.bash is the only script in the repo that is making use of it but you really don't know if users of the repo have external scripts invoking it.  Also, the concern about looping through the entire body of `make-firmware.bash: main()` for each platform I understand but you can easily solve that by moving the content of main to a function and looping through each platform in main invoking the new function.